src/storage/hummock_sdk/src/lib.rs

```diff
@@ -266,3 +266,16 @@ pub fn version_checkpoint_path(root_dir: &str) -> String {
 pub fn version_checkpoint_dir(checkpoint_path: &str) -> String {
     checkpoint_path.trim_end_matches(|c| c != '/').to_string()
 }
+
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Default, Debug)]
+pub struct EpochWithGap(u64);
+
+impl EpochWithGap {
+    pub fn new(epoch_with_gap: u64) -> Self {
```

Comment:
May add method `epoch` and `gap` to return the pure epoch and gap.

Reply:
And also provide a method `new_from_epoch` to explicitly create an `EpochWithGap` from a pure epoch and its  gap is 0. We can do debug_assert in this method to ensure that the epoch is well shifted.

And I think the `new` method is better to pass the `epoch` and `gap` separately, because I think in our system we should't have an `epoch_with_gap` of pure u64 living outside the `EpochWithGap` struct. After we change the method interface, we can ensure this.